$MirOS: ports/net/rrdtool/patches/patch-src_rrd_graph_c,v 1.1 2008/11/12 20:27:50 tg Exp $
--- src/rrd_graph.c.orig	Mon Apr 25 21:03:23 2005
+++ src/rrd_graph.c	Wed Nov 12 23:14:53 2008
@@ -2009,15 +2009,15 @@ horizontal_mrtg_grid(gdImagePtr gif, ima
  xRight = im->xorigin + im->xsize;
 
  if (im->scaledstep / im->magfact * max(abs(im->quadrant),abs(4 - im->quadrant)) <= (double)1)
-    strcpy(labfmt,"%5.2f");
+    strlcpy(labfmt,"%5.2f", sizeof (labfmt));
  else
     sprintf(labfmt,"%%4.%df",1 - (im->scaledstep / im->magfact > (double)10 || ceil(im->scaledstep / im->magfact) == im->scaledstep /im->magfact));
  if (im->symbol != ' ' || im->unit)
-    strcat(labfmt," ");
+    strlcat(labfmt," ",sizeof (labfmt));
  if (im->symbol != ' ')
-    sprintf(labfmt + strlen(labfmt),"%c",im->symbol);
+    snprintf(labfmt + strlen(labfmt),sizeof (labfmt)-strlen(labfmt),"%c",im->symbol);
  if (im->unit)
-    strcat(labfmt,im->unit);
+    strlcat(labfmt,im->unit,sizeof (labfmt));
 
  for (i = 0; i <= 4; i++)
   {y = im->yorigin - im->ysize * i / 4;
@@ -2460,7 +2460,7 @@ copyImage(gdImagePtr gif, char *image, i
       }
       gdImageDestroy(img); 
      }
-     close(fi); 
+     fclose(fi); 
    } else {
      rrd_set_error("Error Opeing %s: %s",image,  strerror(errno));
    }
@@ -2623,7 +2623,7 @@ graph_paint(image_desc_t *im, char ***ca
 
   if (im->bkg_image){	/* background image added first */
     copyImage(gif,im->bkg_image,1);
-    if (rrd_test_error) return(-1);
+    if (rrd_test_error()) return(-1);
   }
   else
    {
@@ -2770,7 +2770,7 @@ graph_paint(image_desc_t *im, char ***ca
 
   if (im->ovl_image) {	/* overlay image added last */
     copyImage(gif,im->ovl_image,0);
-    if (rrd_test_error) return(-1);
+    if (rrd_test_error()) return(-1);
    }
     if (strcmp(im->graphfile,"-")==0) {
 #ifdef WIN32
@@ -3267,7 +3267,7 @@ im.unit = 0; /* do not display unit if n
 	    break;
 	case GF_COMMENT:
 	    if(strlen(&argv[i][argstart])>FMT_LEG_LEN) argv[i][argstart+FMT_LEG_LEN-3]='\0' ;
-	    strcpy(im.gdes[im.gdes_c-1].legend, &argv[i][argstart]);
+	    strlcpy(im.gdes[im.gdes_c-1].legend, &argv[i][argstart], sizeof (im.gdes[im.gdes_c-1].legend));
 	    break;
 	case GF_HRULE:
 	    if(sscanf(
@@ -3291,14 +3291,17 @@ im.unit = 0; /* do not display unit if n
 	    } 
 	    break;
 	case GF_VRULE:
+	  {
+	    uint64_t mock_time;
 	    if(sscanf(
 		&argv[i][argstart],
-		"%lu#%2x%2x%2x:%n",
-		&im.gdes[im.gdes_c-1].xrule,
+		"%llu#%2x%2x%2x:%n",
+		&mock_time,
 		&col_red,
 		&col_green,
 		&col_blue,
 		&strstart) >=  4){
+		im.gdes[im.gdes_c-1].xrule = mock_time;
 		im.gdes[im.gdes_c-1].col.red = col_red;
 		im.gdes[im.gdes_c-1].col.green = col_green;
 		im.gdes[im.gdes_c-1].col.blue = col_blue;
@@ -3313,6 +3316,7 @@ im.unit = 0; /* do not display unit if n
 		return -1;
 	    }
 	    break;
+	  }
 	case GF_STACK:
 	    if(linepass == 0){
 		im_free(&im);
